niworkflows/interfaces/cifti.py

```diff
@@ -403,7 +404,7 @@ def _create_cifti_image(
                 voxel_indices_ijk=vox,
             )
             idx_offset += len(vox)
-            bm_ts = np.column_stack((bm_ts, ts.T))
+            bm_ts = np.column_stack((bm_ts, ts.T)).astype(bold_dtype)
```

Comment:
this will convert cortical + subcortical values to the bold_dtype - basically, is this fine to do?

Reply:
If the original has an on-disk dtype of int16, this is going to induce rounding. Why not just `set_data_dtype` at the end?